master/internal/rm/kubernetesrm/spec.go

```diff
@@ -361,8 +368,24 @@ func (p *pod) configurePodSpec(
 	if podSpec.ObjectMeta.Labels == nil {
 		podSpec.ObjectMeta.Labels = make(map[string]string)
 	}
+	// add determined metadata as pod labels
+	if p.submissionInfo.taskSpec.Owner != nil {
+		// note: owner label will disappear if Owner is somehow nil
```

Comment:
Yeah, I probably over-commented. I removed some, but left comments about missing labels because that's a risk I want to call out / be clear on. Thanks!